config/main.py

```diff
@@ -745,6 +741,10 @@ def config():
     """SONiC command line - 'config' command"""
     if os.geteuid() != 0:
         exit("Root privileges are required for this operation")
+    # Take lock only when config command is executed, to avoid taking lock for
+    # TABs and for -h. Note ? is treated as input python clicks
```

Comment:
So if we use "?", we acquire lock?

Reply:
Yes, because Python-click treats ? as ARG like below
```
admin@lnos-x1-a-fab01:~/test_pc_load_yang$ sudo config bgp shutdown neighbor ?
Usage: config bgp shutdown neighbor [OPTIONS] <ipaddr_or_hostname>
Try "config bgp shutdown neighbor -h" for help.

Error: Could not locate neighbor '?'
admin@lnos-x1-a-fab01:~/test_pc_load_yang$ sudo config save ? -y
Running command: /usr/local/bin/sonic-cfggen -d --print-data > ?
admin@lnos-x1-a-fab01:~/test_pc_load_yang$ ls -l ?
-rw-r--r-- 1 root root 57120 Jul 21 18:48 ?
admin@lnos-x1-a-fab01:~/test_pc_load_yang$
```